src/lightning_lite/plugins/collectives/single_device_collective.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, List, Optional
+
+import torch
+
+from lightning_lite.plugins.collectives.collective import Collective
+from lightning_lite.utilities.types import CollectibleGroup
+
+
+class SingleDeviceCollective(Collective):
+    @property
+    def rank(self) -> int:
+        return 0
+
+    @property
+    def world_size(self) -> int:
+        return 1
+
+    def broadcast(self, tensor: torch.Tensor, *_: Any, **__: Any) -> torch.Tensor:
+        return tensor
+
+    def all_reduce(self, tensor: torch.Tensor, *_: Any, **__: Any) -> torch.Tensor:
+        return tensor
+
+    def reduce(self, tensor: torch.Tensor, *_: Any, **__: Any) -> torch.Tensor:
+        return tensor
+
+    def all_gather(self, tensor_list: List[torch.Tensor], tensor: torch.Tensor, **__: Any) -> List[torch.Tensor]:
+        return [tensor]
+
+    def gather(self, tensor: torch.Tensor, *_: Any, **__: Any) -> Optional[List[torch.Tensor]]:
+        return [tensor]
+
+    def scatter(  # type: ignore[override]
+        self,
+        tensor: torch.Tensor,
+        scatter_list: List[torch.Tensor],  # it doesn't make sense to have a None here for a single device
```

Comment:
it's `Optional` in the base class, because it can be `None` on the ranks, that are not `src`. However, on the single device, there doesn't really exist a rank, which is not `src`, therefore it cannot be `Optional`